lib/Drush/CommandFiles/core/DrupliconCommands.php

```diff
@@ -1,16 +1,34 @@
 <?php
 namespace Drush\CommandFiles\core;
 
+use Consolidation\AnnotatedCommand\CommandData;
+
 class DrupliconCommands {
+  protected $printed = false;
 
   /**
    * Print druplicon as post-command output.
    *
    * @hook post-command *
    * @option $druplicon Shows the druplicon as glorious ASCII art.
    */
-  public function druplicon($result, $argsAndOptions, $annotationData) {
-    if ($argsAndOptions['options']['druplicon']) {
+  public function druplicon($result, CommandData $commandData) {
+    // If one command does a drush_invoke to another command,
+    // then this hook will be called multiple times. Only print
+    // once.  (n.b. If drush_invoke_process passes along the
+    // --druplicon option, then we will still get mulitple output)
+    if ($this->printed) {
+      return;
+    }
+    $this->printed = true;
+    $annotationData = $commandData->annotationData();
+    $commandName = $annotationData['command'];
+    // For some reason, Drush help uses drush_invoke_process to call helpsingle
+    if ($commandName == 'helpsingle') {
+      return;
+    }
+    drush_log(dt('Displaying Druplicon for "!command" command.', array('!command' => $commandName)));
+    if ($commandData->input()->getOption('druplicon')) {
```

Comment:
Robo already has a Config object that can be passed in to a command handler via dependency injection.  I intend to make a system similar to Drush's current procedural system that will just set the option values directly on the $input object, so a command object would only need to have the Config object injected if it needed access to more than just its command args and options.